### PR TITLE
Extract details view

### PR DESCRIPTION
**Pull request by the-daniel-rothig**
### Context

Manage Courses UI needs a preview screen that reflects what a published course will look like in Search And Compare. 

### Changes proposed in this pull request

This extracts the course details view as a ViewComponent in a library that can be used by manage-courses-ui

### Guidance to review

Due to a bug in ASP.NET Core, unfortunately you can't run this with runtime 2.1.400. If you are on this latest version you will need to install the sdk 2.1.300

### CODE REVIEW

File: tests/Unit.Tests/Utils/MarkdownFormatter.Tests.cs [hunk 1]
```diff
@@ -126,46 +126,46 @@ public void NoHtml()
...
-        }
-
-        [Test]
-        public void NotBold2()
-        {
-            var markdown = @"****";
-            var expected = "<p class=\"govuk-body\">****</p>";
-            Assert.That(GetHtml(markdown), Is.EqualTo(expected));
-        }
+//         [Test]
```
Comment (dankmitchell) [inline tests/Unit.Tests/Utils/MarkdownFormatter.Tests.cs]: Do we need to keep commented out code? Can we delete, please?

File: src/Assets/Styles/components/_preview-banner.scss [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.preview-banner
+{
+    @include govuk-font($size: 19, $weight: normal);
+  
+    background-color: #f8f8f8;
```
Comment (dankmitchell) [inline src/Assets/Styles/components/_preview-banner.scss]: Please remove the hex value and add `govuk-colour("grey-4")`

File: src/Assets/Styles/components/_preview-banner.scss [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    padding: 20px;
+    border-top: 5px solid #005ea5;
```
Comment (dankmitchell) [inline src/Assets/Styles/components/_preview-banner.scss]: Please remove hex value with `govuk-colour("blue")`

File: src/Assets/Styles/components/_preview-banner.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+.preview-banner
+{
+    @include govuk-font($size: 19, $weight: normal);
+  
+    background-color: #f8f8f8;
+    padding: 20px;
```
Comment (dankmitchell) [inline src/Assets/Styles/components/_preview-banner.scss]: `padding: govuk-spacing(4)`

File: src/Assets/Styles/components/_preview-banner.scss [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+    border-top: 5px solid #005ea5;
+    margin-top:15px;
```
Comment (dankmitchell) [inline src/Assets/Styles/components/_preview-banner.scss]: `margin-top: govuk-spacing(3)`

File: src/Assets/Styles/components/_missing-section.scss [hunk 6]
```diff
@@ -0,0 +1,6 @@
+.missing-section
+{
+    border: 5px solid #bfc1c3;
```
Comment (dankmitchell) [inline src/Assets/Styles/components/_missing-section.scss]: Replace hex code with `$govuk-border-colour`

File: src/Assets/Styles/components/_missing-section.scss [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+    padding: 15px;
```
Comment (dankmitchell) [inline src/Assets/Styles/components/_missing-section.scss]: `padding: govuk-spacing(3)`

**Comment by the-daniel-rothig** [comment]
[Users submit courses for publication](https://trello.com/c/hzb8hm2o/178-users-submit-courses-for-publication)

**Comment by timabell** [comment]
@the-daniel-rothig could you add the info on sdk version to the readme so to leave a clue for the future somewhere more accessible than a closed PR. What is the failure mode if you don't have the right SDK? Does it fail to build or is it something more subtle. Do you have a link to a github issue on aspnet itself?

**Comment by the-daniel-rothig** [comment]
@timabell I just updated the readme will all the necessary information. Luckily the project refuses to build if the correct SDK is not present, so nothing subtle happening

File: tests/Unit.Tests/Utils/MarkdownFormatter.Tests.cs (hunk 1 above)
Comment (the-daniel-rothig) [reply tests/Unit.Tests/Utils/MarkdownFormatter.Tests.cs]: Done - thanks :+1:

**Pull request merged**